CurrencyConverterService/CurrencyRefreshing.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.EntityFrameworkCore;
+using CurrencyConverterService.Models;
+using System.Linq;
+
+namespace CurrencyConverterService
+{
+    class CurrencyRefreshing
+    {
+        Context dataBase;
+        Logging logging;
+        public CurrencyRefreshing()
+        {
+            this.dataBase = new Context();
+            this.logging = new Logging();
+        }
+        public void AddCurrencyForNBRB(List<CurrencyNBRB> currencysNBRB)
+        {
+            try
+            {
+                const int numberAllCurrencies = 225;
+                IEnumerable<Bank> banks = dataBase.Banks;
+                IEnumerable<Currency> currenciesNBRB = dataBase.Currencies;
+                var bankNBRBId = banks.Where(bank => bank.Name == "NBRB").Select(bank => bank.Id);
+                var amountBankCurrencies = currenciesNBRB.Where(currencie => currencie.BankId == bankNBRBId.FirstOrDefault()).Count();
+                if (amountBankCurrencies == numberAllCurrencies)
+                {
+                    return;
+                }
+                else
+                {
+                    foreach (var currencyNBRB in currencysNBRB)
+                    {
+                        var currency = new Currency();
+                        currency.Name = currencyNBRB.Cur_Abbreviation;
+                        currency.BankId = 1;
+                        dataBase.Currencies.Add(currency);
+                    }
+                    dataBase.SaveChanges();
+                    logging.AddInformation("Currency table updated");
+                }
+            }
+            catch (Exception exception)
```

Comment:
This is potential problem in future. Use concrete exception type like SqlException. Using base Exception will catch all errors and hides incorrect behavior for higher code levels. Using catch (Exception exception) preferred only in high level code.
If you still want to catch base Exception for some handling like logging throw this exception after handling code.